cpp/include/nebula/client/Session.h

```diff
@@ -0,0 +1,71 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#pragma once
+
+#include "nebula/client/Connection.h"
+#include "nebula/data/ResultSet.h"
+
+namespace nebula {
+
+class ConnectionPool;
+
+class Session {
+public:
+    using ExecuteCallback = std::function<void(ResultSet &&)>;
+    using ExecuteJsonCallback = std::function<void(std::string &&)>;
+
+    Session() = default;
+    Session(int64_t sessionId,
+            Connection &&conn,
+            ConnectionPool *pool,
+            const std::string &username,
+            const std::string &password)
+        : sessionId_(sessionId),
+          conn_(std::move(conn)),
+          pool_(pool),
+          username_(username),
+          password_(password) {}
+    Session(Session &&session)
+        : sessionId_(session.sessionId_),
+          conn_(std::move(session.conn_)),
+          pool_(session.pool_),
+          username_(std::move(session.username_)),
+          password_(std::move(session.password_)) {
+        session.sessionId_ = -1;
+        session.pool_ = nullptr;
+    }
+    ~Session() {
+        release();
+    }
+
+    ResultSet execute(const std::string &stmt);
+
+    void asyncExecute(const std::string &stmt, ExecuteCallback cb);
```

Comment:
need to add cb is `void asyncExecute(const std::string &stmt, void(*ExecCallback)(ResultSet*), void *context)` because gcc4.8 is not supported std::function

Reply:
The gcc4.8 API support std::function if specify c++11.